docs/readmes/orc8r/architecture_modularity.md

```diff
@@ -0,0 +1,99 @@
+---
+id: architecture_modularity
+title: Modularity
+hide_title: true
+---
+
+# Modularity
+
+The Orchestrator follows a modular design, supporting domain-specific applications under an extendable service mesh architecture. This
+document describes the Orchestrator's service mesh architecture motivations, implementation, and how to extend Orchestrator for your own
+domain-specific purposes.
+
+## Motivation
+
+The Magma platform aims to provide unified access network management, across domains. Operators should be able to use the same,
+familiar Orchestrator interface to provision and monitor deployments targeting use-cases from fixed wireless access, federated fixed
+wireless access, private LTE, carrier Wi-Fi, etc.
+
+To achieve this goal, the Orchestrator provides a set of core, domain-agnostic features, which are then extended by domain-specific
+functionality on a per-deployment basis. This logic-based extension occurs completely at runtime -- that is, while we provide Helm charts
+targeting a default set of use-cases, extending an Orchestrator deployment is, practically, as simple as spinning up a new service within
```

Comment:
Might be worth mentioning on why spinning up a new service within Orc8r is a useful feature, for e.g. you can seamlessly add new functionality to Orc8r without necessarily upgrading the Orc8r, Fixing critical bugs by simply installing new helm chart without affecting existing deployment.